File: server/src/main/scala/producer/message_generator/RandomBytesGenerator.scala
package producer.message_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class RandomBytesGenerator(
    minBytes: Int,
    maxBytes: Int
)

object RandomBytesGenerator:
    def fromPb(v: pb.RandomBytesGenerator): RandomBytesGenerator =
        RandomBytesGenerator(
            minBytes = v.minBytes, 
            maxBytes = v.maxBytes
        )

    def toPb(v: RandomBytesGenerator): pb.RandomBytesGenerator =
        pb.RandomBytesGenerator(
            minBytes = v.minBytes, 
            maxBytes = v.maxBytes
        )

File: ui/components/ui/ProducerSession/message-generator/data-generators/bytes/random-bytes-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type RandomBytesGenerator = {
  type: 'random-bytes-generator',
  minBytes: number,
  maxBytes: number
};

export function randomBytesGeneratorFromPb(v: pb.RandomBytesGenerator): RandomBytesGenerator {
  return {
    type: 'random-bytes-generator',
    minBytes: v.getMinBytes(),
    maxBytes: v.getMaxBytes()
  };
}

export function randomBytesGeneratorToPb(v: RandomBytesGenerator): pb.RandomBytesGenerator {
  const message = new pb.RandomBytesGenerator();
  message.setMinBytes(v.minBytes);
  message.setMaxBytes(v.maxBytes);

  return message;
}
